middleware/swagger-router.js

```diff
@@ -77,26 +78,39 @@ var handlerCacheFromDir = function (dirOrDirs) {
       var controller;
 
       if (file.match(jsFileRegex)) {
-        controller = require(path.resolve(path.join(dir, controllerName)));
+        try {
+          controller = require(path.resolve(path.join(dir, controllerName)));
 
-        debug('    %s%s:',
-              path.resolve(path.join(dir, file)),
-              (_.isPlainObject(controller) ? '' : ' (not an object, skipped)'));
+          debug('    %s%s:',
+                path.resolve(path.join(dir, file)),
+                (_.isPlainObject(controller) ? '' : ' (not an object, skipped)'));
 
-        if (_.isPlainObject(controller)) {
-          _.each(controller, function (value, name) {
-            var handlerId = controllerName + '_' + name;
+          if (_.isPlainObject(controller)) {
+            _.each(controller, function (value, name) {
+              var handlerId = controllerName + '_' + name;
 
-            debug('      %s%s',
-                  handlerId,
-                  (_.isFunction(value) ? '' : ' (not a function, skipped)'));
+              debug('      %s%s',
+                    handlerId,
+                    (_.isFunction(value) ? '' : ' (not a function, skipped)'));
 
-            // TODO: Log this situation
+              // TODO: Log this situation
 
-            if (_.isFunction(value) && !handlerCache[handlerId]) {
-              handlerCache[handlerId] = value;
-            }
-          });
+              if (_.isFunction(value) && !handlerCache[handlerId]) {
+                handlerCache[handlerId] = value;
+              }
+            });
+          }
+        } catch (err) {
+          // TODO: Change this logging to match the above when the overall
+          // logging piece is done.
+          var message = util.format('%s - Module loader error: %s',
+              path.resolve(path.join(dir, file)),
+              err);
+          debug('    %s', message);
+          if (console) {
```

Comment:
You're probably right, you can just chop that check out - Its just a person coding habit - I generally always prefix it because you never know if code has a life beyond what was originally written.